workspace.go

```diff
@@ -969,3 +986,47 @@ func (s *workspaces) UpdateRemoteStateConsumers(ctx context.Context, workspaceID
 
 	return s.client.do(ctx, req, nil)
 }
+
+type WorkspaceTagsOptions struct {
+	Tags []TagOptions `jsonapi:"relation,tags,data,omitempty`
+}
+
+func (o WorkspaceTagsOptions) valid() error {
+	for _, s := range o.Tags {
+		if s.Name == nil && s.Id == nil {
+			return ErrMissingTagIdentifier
+		}
+	}
+
+	return nil
+}
+
+func (s *workspaces) AddTags(ctx context.Context, workspaceID string, options WorkspaceTagsOptions) error {
+	if err := options.valid(); err != nil {
+		return err
+	}
+
+	u := fmt.Sprintf("workspaces/%s/relationships/tags", url.QueryEscape(workspaceID))
+	req, err := s.client.newRequest("POST", u, &options)
+
+	if err != nil {
+		return err
+	}
+
+	return s.client.do(ctx, req, nil)
+}
+
+func (s *workspaces) DeleteTags(ctx context.Context, workspaceID string, options WorkspaceTagsOptions) error {
```

Comment:
nit: `RemoveTags` might be more appropriate because we're not deleting the tag resource. we're just removing it from this workspace.